File: algorithms/python/bt-inorder.py
class BinaryNode:
    def __init__(self) -> None:
        self.value: int
        self.left: BinaryNode
        self.right: BinaryNode


def walk(curr: BinaryNode | None, path: list[int]) -> list[int]:
    if not curr:
        return path
    walk(curr.left, path)
    path.append(curr.value)
    walk(curr.right, path)
    return path


def inorder_search(head: BinaryNode) -> list[int]:
    return walk(head, [])

File: algorithms/typescript/BT-inorder.ts
import { BinaryNode } from "./BT-preorder";

function inorder_walk(curr: BinaryNode<number>, path: number[]): number[] {
  if (!curr) {
    return path;
  }
  inorder_walk(curr.left, path);
  path.push(curr.value);
  inorder_walk(curr.right, path);
  return path;
}
function inorder_search(head: BinaryNode<number>, path: number[]): number[] {
  return inorder_walk(head, []);
}
